api-server/src/server/boot/donate.js

```diff
@@ -26,6 +28,22 @@ export default function donateBoot(app, done) {
     });
   }
 
+  async function handleStripeCardDonation(req, res) {
+    return createStripeCardDonation(req, res, stripe, app).catch(err => {
+      log(err.message);
+      if (
+        err.type === 'LargeNumbersOfDonations' ||
+        err.type === 'HighDonationFrequency' ||
+        err.type === 'InvalidRequest'
+      ) {
+        return res.status(500).send({ error: err.message });
+      }
```

Comment:
Mostly for my benefit, but I am asking on here in case others are also curious:

Why do we handle these specific error types differently?

Reply:
Also, I do not think `500` is an accurate error in this case?

Reply:
There are there in case we need to show relevant messages to the user.

At this point any error would cause this route to throw causing the client to show a default message.
You are right, it should probably be 400.

Reply:
I will just return 402 here for isDonating users